processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -341,31 +341,21 @@ public File merge(
 
     final List<String> mergedDimensions = getMergedDimensions(indexes);
 
-    final List<String> mergedMetrics = Lists.transform(
-        mergeIndexed(
-            Lists.newArrayList(
-                FunctionalIterable
-                    .create(indexes)
-                    .transform(
-                        new Function<IndexableAdapter, Iterable<String>>()
-                        {
-                          @Override
-                          public Iterable<String> apply(@Nullable IndexableAdapter input)
-                          {
-                            return input.getMetricNames();
-                          }
-                        }
-                    )
-            )
-        ),
-        new Function<String, String>()
-        {
-          @Override
-          public String apply(@Nullable String input)
-          {
-            return input;
-          }
-        }
+    final List<String> mergedMetrics = mergeIndexed(
```

Comment:
Please do this in another PR, it makes life easier if a commit contains a set of related changes.